spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationPropertySource.java

```diff
@@ -249,6 +272,56 @@ private Object createFeature(ConfigurationSetting item) throws IOException {
                 } else if (!featureItem.getEnabled()) {
                     return false;
                 }
+                mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```

Comment:
It's not recommended been change the mapper configuration in runtime. It's not thread-safe beyond what any error during this code before the return to false could leave this the object in an inconsistent state.  Maybe to have an objectMapper object with these characteristics?